CRM/Mailing/BAO/MailingJob.php

```diff
@@ -836,6 +836,49 @@ public static function cancel($mailingId) {
     }
   }
 
+  /**
+   * Pause a mailing
+   */
+  public static function pause($mailingID) {
+    $sql = "
+      UPDATE civicrm_mailing_job
+      SET status = 'Paused'
+      WHERE mailing_id = %1
+      AND is_test = 0
+      AND status IN ('Scheduled', 'Running')
+    ";
+    CRM_Core_DAO::executeQuery($sql, array(1 => array($mailingID, 'Integer')));
+
+    CRM_Core_Session::setStatus(ts('The mailing has been paused.'), ts('Paused'), 'success');
```

Comment:
I'd vote to put the `setStatus()` calls in the form layer (`CRM/Mailing/Page/Browse.php`).

For example, if one were writing a replacement UI or doing a script which called this action, then the messaging would be different, but the base logic should be the same.

Reply:
Agreed. Migrated `setStatus()` to Browse.php 👍